src/components/Dialog/PDialog.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+  <dialog ref="dialog" class="p-dialog" @click="onClick">
+    <div class="p-dialog__content">
+      <template v-if="open">
+        <slot v-bind="{ close }" />
+      </template>
+    </div>
+  </dialog>
+</template>
+
+<script lang="ts" setup>
+  import { onMounted, onUnmounted, ref, watch } from 'vue'
+
+  const props = defineProps<{
+    open: boolean,
+    modal?: boolean,
+    autoClose?: boolean,
+  }>()
+
+  const emit = defineEmits<{
+    (event: 'update:open', value: boolean): void,
+  }>()
+
+  const dialog = ref<HTMLDialogElement>()
+
+  function show(): void {
+    if (!dialog.value) {
+      return
+    }
+
+    if (props.modal) {
+      dialog.value.showModal()
+      return
+    }
+
+    dialog.value.show()
+  }
+
+  function close(): void {
+    if (!dialog.value) {
+      return
+    }
+
+    dialog.value.close()
+  }
+
+  function syncOpenValue(): void {
+    if (!dialog.value) {
+      return
+    }
+
+    emit('update:open', dialog.value.open)
+  }
+
+  function onClick(event: MouseEvent): void {
+    if (!props.autoClose) {
```

Comment:
I would have expected `auto-close` to trigger close on any outside click? Isn't that how it works on a-modal and a-popover?

Reply:
I think that's how this is working. If you add the `auto-close` prop if you click anywhere on the backdrop (technically anywhere outside of the `p-dialog__content` div) it'll call the `close` method.

Reply:
interesting. I'm surprised that the click handler on `dialog` gets triggered by anything on the page.

I also noticed that your check for clicks inside the dialog only checks if the target === the dialog element. Wouldn't it be better to check if `dialog.value.contains(event.target)`?

Reply:
hmm so when a dialog is opened as a modal (using `showModal`) there is a `::backdrop` that covers the entire page. So anywhere you can click you trigger a click on the dialog element. So that's why I'm checking that the click is the dialog and not one of its children. (checking dialog.contains would mean clicking on the content of the dialog would close it as well). BUT that doesn't work if its not a modal, so I'll need to account for that. Good catch!

Reply:
@stackoverfloweth fixed!